src/http/api/views/blocks.view.ts

```diff
@@ -0,0 +1,46 @@
+import type { Knex } from 'knex';
+import type { BlockedDomainDTO, MinimalAccountDTO } from '../types';
+
+export class BlocksView {
+    constructor(private readonly db: Knex) {}
+
+    async getBlockedAccounts(accountId: number): Promise<MinimalAccountDTO[]> {
+        const results = await this.db('blocks')
+            .select([
+                'accounts.ap_id',
+                'accounts.name',
+                'accounts.username',
+                'accounts.avatar_url',
+                'domain_blocks.domain',
+            ])
+            .innerJoin('accounts', 'accounts.id', 'blocks.blocked_id')
+            .leftJoin(
+                'domain_blocks',
+                'domain_blocks.domain',
+                'accounts.domain',
+            )
+            .where('blocks.blocker_id', accountId);
+
+        return results.map((result) => ({
+            id: result.ap_id,
+            apId: result.ap_id,
+            name: result.name || '',
+            handle: `${result.username}@${new URL(result.ap_id).host}`,
```

Comment:
Could add `accounts.domain` to the select to avoid doing a `new URL` over ap_id here?

Reply:
Great shout - we can probably do that in a lot of places!